fconfig/config.go

```diff
@@ -0,0 +1,105 @@
+package fconfig
+
+import (
+	"os"
+	"regexp"
+
+	"github.com/Flahmingo-Investments/helpers-go/ferrors"
+	"github.com/Flahmingo-Investments/helpers-go/gcpauth"
+	"github.com/joho/godotenv"
+	"github.com/spf13/viper"
+)
+
+var secretRegex = regexp.MustCompile(`^gSecret://(?P<Path>.+)`)
+var envFileRegex = regexp.MustCompile(`\.env$`)
+
+// LoadConfigViper loads the configuration from a given file.
+// For environment variables use the following format int the FILE. `${<Env var>}`
+// For Google secrets, use the following format.
+// gSecret://projects/<projectId>/secrets/<secretName>/versions/<version id or 'latest'>
+// In order to parse the config, run v.Unmarshal(<config struct>), and ensure all your properties are exported/public
+func LoadConfigViper(files []string) (*viper.Viper, error) {
+	var err error
+
+	v := viper.New()
+	v.SetEnvPrefix("")
+	v.AutomaticEnv()
+
+	firstFile := true
+	for _, file := range files {
+		if envFileRegex.MatchString(file) {
+			err = loadEnv(file)
+			if err != nil {
+				return nil, err
+			}
+		} else if firstFile {
+			v.SetConfigFile(file)
+			firstFile = false
+		} else {
+			v.AddConfigPath(file)
+		}
+	}
+
+	err = v.ReadInConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	for _, key := range v.AllKeys() {
+		val := v.GetString(key)
+
+		v.Set(key, os.ExpandEnv(val))
+
+		if secretRegex.MatchString(val) {
+			secretPath := getSecretPath(val)
+			secret, err := gcpauth.GetSecretByName(secretPath)
+			if err != nil {
+				return nil, err
+			}
+			v.Set(key, secret)
```

Comment:
I moved it into a local function. 
The gcp auth library is a generic library for getting secrets, don't want to have it parse a regex.
The only reason we are doing the path construct here, is because it needs to stand out.